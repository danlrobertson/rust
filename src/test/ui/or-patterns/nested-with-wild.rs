// run-pass
#![feature(or_patterns)]
//~^ WARN the feature `or_patterns` is incomplete and may cause the compiler to crash

#[derive(Debug)]
pub enum Test {
    Foo(usize, Option<usize>),
    Bar,
    Baz
}

pub fn test(x: Option<Test>) -> bool {
    match x {
        Some(Test::Bar | Test::Foo(_, Some(_))) => true,
        Some(_) => false,
        _ => false,
    }
}

fn main() {
    assert!(test(Some(Test::Foo(42, Some(1)))));
    assert!(test(Some(Test::Foo(255, Some(255)))));
    assert!(test(Some(Test::Bar)));
    assert!(!test(Some(Test::Baz)));
    assert!(!test(None));
}
