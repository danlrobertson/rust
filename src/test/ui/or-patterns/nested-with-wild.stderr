warning: the feature `or_patterns` is incomplete and may cause the compiler to crash
  --> $DIR/nested-with-wild.rs:2:12
   |
LL | #![feature(or_patterns)]
   |            ^^^^^^^^^^^

